tests/test_operation.py

```diff
@@ -1317,10 +1317,16 @@ class TestOperationDerivative:
 
     def test_no_generator_raise(self):
         """Tests if the function raises an exception if the input operation has no generator"""
-        op = qml.Rot(0.1, 0.2, 0.3, wires=0)
+
+        class CustomOp(qml.operation.Operation):
+            num_wires = 1
+            num_params = 1
+
+        op = CustomOp(0.5, wires=0)
```

Comment:
Out of curiousity: Why was this change required? :)

Reply:
@cvjjm suggested (very wisely) to move the `num_params` validation _before_ getting the generator. So this test started raising 'Generator only supports single param operations' instead!

I couldn't think of another gate that would fit the test, and probably better practise to have test data designed for the test.

Reply:
Thanks for the explanation :)